src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -128,6 +129,66 @@ public static async Task ZipArchiveEntry_InvalidUpdate(string zipname)
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Fix not shipped for .NET Framework.")]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/32316", TestRuntimes.Mono)]
+        public static void ZipArchiveEntry_AppContext_Switch_SuppresHeaderValidation(bool suppressValidation)
+        {
+            RemoteExecutor.Invoke((suppressString) =>
+            {
+                bool suppress = bool.Parse(suppressString);
+                MemoryStream stream = populateStream().Result;
+                PatchDataRelativeToFileName(Encoding.ASCII.GetBytes(s_tamperedFileName), stream, 8);  // patch uncompressed size in file header
+                using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+                {
+                    // Set the AppContext Switch
+                    AppContext.SetSwitch("System.Compression.ZipArchiveEntry.SuppressHeaderValidation", suppress);
+                    ZipArchiveEntry e = archive.GetEntry(s_tamperedFileName);
+                    if (suppress)
+                    {
+                        using (MemoryStream ms = new MemoryStream())
+                        using (Stream source = e.Open())
+                        {
+                            source.CopyTo(ms);
+                            Assert.Equal(ms.Position, ms.Length);  // Just making sure it was readable
+                        }
+                    }
+                    else
+                    {
+                        Assert.Throws<InvalidDataException>(() => e.Open());
+                    }
+                }
+            }, suppressValidation.ToString()).Dispose();
+        }
+
+        [Fact]
+        public static async Task LargeArchive_DataDescriptor_Read_NonZip64_FileLengthGreaterThanIntMax()
+        {
+            MemoryStream stream = await LocalMemoryStream.readAppFileAsync(strange("fileLengthGreaterIntLessUInt.zip"));
+
+            using (ZipArchive archive = new ZipArchive(stream, ZipArchiveMode.Read))
+            {
+                ZipArchiveEntry e = archive.GetEntry("large.bin");
+
+                Assert.Equal(3_600_000_000, e.Length);
+                Assert.Equal(3_499_028, e.CompressedLength);
+
+                using (Stream source = e.Open())
+                {
+                    byte[] buffer = new byte[s_bufferSize];
+                    int read = source.Read(buffer, 0, buffer.Length);   // We don't want to inflate this large archive entirely 
+                                                                        // just making sure it read successfully 
+                    Assert.Equal(s_bufferSize, read);
+                    foreach (byte b in buffer)
+                    {
+                        Assert.Equal((byte)'0', b); // the file should be all "0"s 
```

Comment:
Is this slow - a method call for every byte? 10240 calls to Assert.Equal.

I do not know of a managed equivalent for memcmp, but you could still take the assert out of the loop. For example, just make your loop sum the bytes, and assert the sum is zero.

Reply:
`Assert.Equal` can compare sequences, but you would need to fake a sequence of 0's to use it. I would just sum.

Reply:
On my machine this test runs in 20 ms. Removing this check completely takes it down to around 10 ms. Do you think it is worth it to save 10 ms?

> For example, just make your loop sum the bytes, and assert the sum is zero.

Note that the byte isn't `0x00`. The value is the ASCII character `0`'s byte value - `0x30`.

Reply:
Fair enough - fine. And I misread

Reply:
`Assert.IsTrue(buffer.All(b => b == 0))` could be a clean alternative.

Reply:
@gspp I guess that would be `Assert.IsTrue(buffer.All(b => b == `0`))`